### PR TITLE
sttr: add version 0.2.12

### PR DESCRIPTION
**Pull request by abhimanyu003**
- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tech189** [comment]
Hi! Thanks for your PR, please could you follow the contribution guide and give some details about the popularity of this program? Next time it would be great if you could create an issue for discussion first :) At a glance it seems a popular program but I will take a closer look tomorrow. Could you also please fix the manifest so that it passes the checks?

File: bucket/sttr.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+    "version": "0.2.12",
+    "description": "cross-platform, cli app to perform various operations on string",
```
Comment (rashil2000) [inline bucket/sttr.json:3]:
```suggestion
    "description": "Command line application to perform various operations on strings",
```

File: bucket/sttr.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    "homepage": "https://github.com/abhimanyu003/sttr",
+    "license": {
+        "identifier": "MIT",
+        "url": "https://github.com/abhimanyu003/sttr/blob/main/LICENSE"
+    },
```
Comment (rashil2000) [inline bucket/sttr.json:8]:
```suggestion
    "license": "MIT",
```

File: bucket/sttr.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        "architecture": {
+            "64bit": {
+                "url": "https://github.com/abhimanyu003/sttr/releases/download/v$version/sttr_$version_windows_amd64.tar.gz"
+            },
+            "32bit": {
+                "url": "https://github.com/abhimanyu003/sttr/releases/download/v$version/sttr_$version_windows_386.tar.gz"
+            }
+        },
+        "hash": {
+            "url": "$baseurl/checksums.txt"
```
Comment (rashil2000) [inline bucket/sttr.json:31]:
```suggestion
            "url": "$baseurl/sttr_$version_checksums.txt"
```

**Review by rashil2000 — CHANGES_REQUESTED** [review]

**Comment by rashil2000** [comment]
https://github.com/ScoopInstaller/Main/pull/2859#issuecomment-955644109

**Comment by abhimanyu003** [comment]
Hello, @tech189 and @rashil2000 thanks so much for your feedback and suggestions.
It's command-line software so I thought it will be allowed. 
Few points: It's not on 500 starts yet, but it was trending #3 on ProductHunt https://www.producthunt.com/posts/sttr 
It's not downloading anything or opening up any ports or runs in the background, security is not an issue here.

I tried to fix the manifest as per review comments but somehow it keeps on failing. 

Please let me know your thoughts.

**Comment by rashil2000** [comment]
There's good reason why we **insist** on contributors opening issues first for any PR, so that such discussions can be had.

> It's command-line software so I thought it will be allowed.

It's not the only [criterion](https://github.com/ScoopInstaller/scoop/wiki/Criteria-for-including-apps-in-the-main-bucket).

> It's not downloading anything or opening up any ports or runs in the background, security is not an issue here.

We know, we can all see the source code there :)

The popularity/age criterion is present to establish a base level of trust. The release artifacts an author puts on their GitHub repo can be modified, even if they're released through an automated process (like GitHub Actions).

> I tried to fix the manifest as per review comments but somehow it keeps on failing.

Our CI is broken as of now, you can ignore it.

> Please let me know your thoughts.

If you insist, you can shift this PR to Extras bucket.

**Comment by abhimanyu003** [comment]
Sure I will look forward to adding it there. Currently closing this PR.

**Comment by tech189** [comment]
If it doesn’t make it into the Extras bucket, it’s very easy to set up a bucket for your own apps. It would only be one extra cmd for your users to run.

**Comment by rashil2000** [comment]
> Sure I will look forward to adding it there.

Hey @abhimanyu003 :) 
Any updates?

**Comment by tech189** [comment]
I added it to my bucket in the meantime, just run the following:
```
scoop bucket add tech189 https://github.com/tech189/tech189-bucket
```
Then install with:
```
scoop install sttr
```

**Comment by rashil2000** [comment]
Interesting!

I'd like to add that we have made a template repository for making personal buckets very easily - https://github.com/ScoopInstaller/BucketTemplate - which has [several advantages](https://github.com/ScoopInstaller/Scoop/wiki/Buckets#using-template).

**Comment by tech189** [comment]
Oh, I didn't know about that template! I'm going to try migrating my bucket to it by copying what my bucket doesn't have.

**Comment by tech189** [comment]
@rashil2000 just to let you know, I've successfully migrated and set up CI tests in my bucket!

**Comment by rashil2000** [comment]
Nice! 😃

**Pull request closed without merging**